src/view/components/common/login-as/login-as.ts

```diff
@@ -17,20 +19,29 @@ import {Router} from '@ngrx/router';
 export class LoginAsComponent {
     @Output() cancel = new EventEmitter<>();
 
-    private userLists: Array<User>;
+    private i18nKeys: Array<string> = ['change', 'cancel', 'password'];
+    private i18nMessages = {};
+    private messageMapSubscription;
     private needPassword: boolean = false;
+    private userLists: Array<User>;
 
-    constructor(private loginService: LoginService, private router: Router,
-                @Inject('dotcmsConfig') private dotcmsConfig: DotcmsConfig) {
+    constructor(private loginService: LoginService, private router: Router, private messageService: MessageService) {
+        this.messageMapSubscription = this.messageService.getMessages(this.i18nKeys).subscribe(res => {
+            this.i18nMessages = _.pick(res, this.i18nKeys);
```

Comment:
We could create a BaseComponent class and move this code to it, something like:

class BaseComponent{
   constructor(i18nKeys:string[], messageService: MessageService){
       this.messageService.getMessages(i18nKeys).subscribe(res => {
          ....
       });
   }
} 

and to use it:

class MyComponent extends BaseComponents{
     constructor(){
          super(['key-1', 'key-2']);
     }
}